### ISSUE TITLE
MAX and STACK Buttons doesn't work. 

### ISSUE THREAD

**Issue by Galthariel** [Sonaza/Vendorer#33]
## Expected Behavior

That at Vendor all 4 Buttons options work (MAX, STACK, OK and CANCEL).

## Actual Behavior

At Vendor only the two bottom buttons work (OK and CANCEL)

## Steps to Reproduce the Problem

  1. Go to any Vendor which includes the Yak Mount and try to buy a Stack of anything

Tried everything I could think of. Disabled all addons literally and left !Vendorer alone enabled and the issue persists. Deleted the !Vendorer.lua in my Saved Variables (also the bak file) and the issue remains. Deleted the whole Vendorer folder and replace it with the one in here and didn't work also.

Thanks

**Comment by sebid** [comment]
I can confirm that with my Vendorer install the buttons are properly working.

You can try fully backup-ing your WTF folder, removing it, and letting your settings re-initialize, checking if you can click the buttons.
Secondly, try hovering over the Max button, for example to check if it shows the maximum stack size it can get.
![image](https://user-images.githubusercontent.com/761115/201306349-ae61d21e-88bf-4032-a010-c147ab06306c.png)

Thirdly, try adding a few debug prints, ex. VendorerStackSplitFrame L422,L424,L426 `print("set max ", self.split);`, to test if the function to set a max size is fired.

Fourthly, what locale do you use?

**Comment by Galthariel** [comment]
1st - Did a full backup of my WTF Folder, and only copied Vendorer as an addon. Same issue
2nd - When I mouse over both, MAX and STACK  they do highlight, and the tooltip shows I can buy only 0?. (seen screenshot below)
3rd - I couldn't get that command to work in Chat or most likely I don't know what should I do to use the debug print
4th - US Locale
![2022-11-11_22-11-05](https://user-images.githubusercontent.com/51491805/201449604-19c19aa5-5e04-46eb-92ec-aeec60714143.png)

**Comment by Galthariel** [comment]
There you go

![2022-11-12_09-58-10](https://user-images.githubusercontent.com/51491805/201475617-96dd2297-fb14-40a3-b7fc-5e793eba1eb8.png)

**Comment by sebid** [comment]
Yeah, this is weird, something is not being initialized well.
VendorerStackSplitFrame.lua
L539-540
	self.canFitStacks   = Addon:GetFreeBagSlotsForItem(itemLink);
	self.canFitItems    = self.canFitStacks * maxStack;

That variable might return 0 for you for a reason. Might be a change in Blizzard's API with bags, something to test. Do you have combine bags or using any bag addon?

**Comment by Galthariel** [comment]
yes, I do have combined bags and use Bagnon. Just remember that I tried with any addons loaded (only Vendorer) and as well with UI reset after backup my WTF folder.

**Comment by sebid** [comment]
But have you tried un-combining bags and reloading?

**Comment by Galthariel** [comment]
I have not...will try that and let you know. AT work so it will be only later.

**Comment by sebid** [comment]
Hmm, I was unable to duplicate the issue by playing with Combined Bags on/off, but I was able to find a quirk with how bag slots are counted.
It appears that it's ignoring any extra bag I have, except for the base backpack.
![image](https://user-images.githubusercontent.com/761115/201676376-ca1624a2-9b14-4d2e-a6e2-2dc528892ad3.png)
Eg. 28 slots, but only 200x20 shown in vendor.
![image](https://user-images.githubusercontent.com/761115/201676479-61d24797-0fe4-4c74-b46f-3eea0bf1e6d2.png)

Found a function that was changed.

VendorerStackSplitFrame.lua
L633		local bagItemLink = GetInventoryItemLink("player", 19 + bagID);
https://wowpedia.fandom.com/wiki/InventorySlotId  -> it is now ContainerIDToInventoryID()

![image](https://user-images.githubusercontent.com/761115/201678084-c503508e-bba7-4437-859c-0bdfb9d647ba.png)

**Comment by sebid** [comment]
Please try this version instead:
https://github.com/sebid/Vendorer/archive/refs/heads/master.zip

**Comment by Galthariel** [comment]
TY...will test later. As a side note I have the 4 extra bags from the Authenticator. Don't know if could be the issue

**Comment by sebid** [comment]
I also do, should not be a problem as Blizzard returns 16 or 20.

**Comment by Galthariel** [comment]
Fixed :-)

![2022-11-14_18-52-35](https://user-images.githubusercontent.com/51491805/201776265-30f0942a-e38b-4200-b543-e86b22727389.png)

![2022-11-14_18-54-31](https://user-images.githubusercontent.com/51491805/201776290-15806b49-bf0f-4f47-ad04-bf79de28202a.png)


Thanks much. I knew wasn't me :-)

**Comment by sebid** [comment]
Awesome :)

**Comment by Galthariel** [comment]
Should I keep using the one you sent me or download any other you may have with consolidated fixes?

**Comment by sebid** [comment]
They're the same :)

**Comment by Galthariel** [comment]
Ok. Thanks for all the help :-)
